erdos/utils.py

```diff
@@ -114,3 +117,13 @@ def log_graph_to_dot_file(filename, nodes, edges):
     # dot footer
     dot_file.write("}")
     dot_file.close()
+
+
+def setup_recorder(name, record_type):
+    filename = '{}.pkl'.format(name)
+    if '/' in name:  # Slash will be mistaken as path path
+        filename = '_'.join(filename.split('/'))
+    f = open(filename, "wb")
+    record_info = "record name: {}; record type: {}".format(name, str(record_type))
+    pickle.dump(record_info, f)
```

Comment:
One issue here is that we never close the recorded file, which means we may not record the last few messages. Instead, we should ensure that the file is closed when the stream is destructed. IMHO, given that the recorded file is strictly a property of the stream, this whole setup_recorder method should be in the data_stream class, and the file should be a field.